### PR TITLE
update cosmpy dependency to version 0.5

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](https://github.com/fetchai/agents-aea/blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](https://github.com/fetchai/agents-aea/blob/master/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`scripts/generate_ipfs_hashes.py`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](https://github.com/fetchai/agents-aea/blob/master/develop-image/README.md#publish) 
      and [`aea-user`](https://github.com/fetchai/agents-aea/blob/master/develop-image/user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: scripts/check_imports_and_dependencies.py [hunk 1]
```diff
@@ -60,6 +58,10 @@ class DependenciesTool:
     @staticmethod
     def get_package_files(package_name: str) -> List[Path]:
         """Get package files list."""
+        from pip._internal.commands.show import (  # type: ignore  # noqa  # pylint: disable=import-outside-toplevel
+            search_packages_info,
+        )
+
```
Comment (5A11) [inline scripts/check_imports_and_dependencies.py:64]: Why moved here?

File: tox.ini [hunk 2]
```diff
@@ -42,23 +42,28 @@ deps =
...
+	py-sr25519-bindings==0.1.4
+    cosmpy>=0.5.0,<0.6.0
 
 
 commands =
 	; for some reason tox installs aea without respect to the dependencies version specified in setup.py. at least in CI env
 	; so install current aea in a normal way
 	pip install {toxinidir}[all]
 	python -m pip install --no-deps file://{toxinidir}/plugins/aea-ledger-ethereum
+    python -m pip install --no-deps file://{toxinidir}/plugins/aea-ledger-ethereum
```
Comment (5A11) [inline tox.ini:66]: Duplicate?

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -16,16 +16,18 @@ jobs:
...
           sudo apt-get update --fix-missing
+          sudo apt-get install libmbedtls-dev
           sudo apt-get autoremove
           sudo apt-get autoclean
-          pip install pipenv
+          sudo pip install pipenv
       - name: Pipenv install requirements and check it can be locked
         run: |
-          pipenv install --dev --skip-lock --clear
+          pip install "cosmpy>=0.5.0,<0.6.0"
```
[the remark this answers is not in the document]
Comment (solarw) [reply .github/workflows/workflow.yml:29]: fixed

File: .github/workflows/workflow.yml [hunk 4]
```diff
@@ -16,16 +16,18 @@ jobs:
       - uses: actions/checkout@master
       - uses: actions/setup-python@v3
         with:
-          python-version: 3.7
+          python-version: 3.9
       - name: Install dependencies (ubuntu-latest)
         run: |
           sudo apt-get update --fix-missing
+          sudo apt-get install libmbedtls-dev
```
[the remark this answers is not in the document]
Comment (solarw) [reply .github/workflows/workflow.yml:23]: new dep for cosmpy requires this library

File: scripts/check_imports_and_dependencies.py (hunk 1 above)
Comment (solarw) [reply scripts/check_imports_and_dependencies.py:64]: did a small fix mfor this script, was a conflict with some import order

File: tox.ini (hunk 2 above)
Comment (solarw) [reply tox.ini:66]: fixed

**Pull request merged**